market-test/src/test/java/com/axonivy/market/test/ValidateRepoTest.java

```diff
@@ -71,11 +79,58 @@ private void assertValidMetaJson(Path path)
           installer = toJsonObject(path.resolve(include));
         }
         JSONObjectAssert.assertThat(installer, metaPath)
-                .requireStringPropertyWithMinLength("id", 5);
+          .requireStringPropertyWithMinLength("id", 5);
+
+        var id = installer.getString("id");
+        if ("maven-dependency".equals(id))
+        {
+          checkMavenArtifactsOfInstaller(installer, "dependencies");
+        }
+        if ("maven-import".equals(id))
+        {
+          checkMavenArtifactsOfInstaller(installer, "projects");
+        }
       }
     }
   }
+  
+  private void checkMavenArtifactsOfInstaller(JSONObject installer, String depsId)
+  {
+    var data = installer.getJSONObject("data");
+    var deps = data.getJSONArray(depsId);
+    
+    var repoUrl = "https://repo1.maven.org/maven2";
+    if (data.has("repositories"))
+    {
+      var repos = data.getJSONArray("repositories");
+      repoUrl = repos.getJSONObject(0).getString("url");
+    }
+    
+    for (var k = 0; k < deps.length(); k++)
+    {
+      var dep = deps.getJSONObject(k);
+      var groupId = dep.getString("groupId");
+      var artifactId = dep.getString("artifactId");
+      checkArtifact(repoUrl, groupId, artifactId);
+    }
+  }
 
+  private void checkArtifact(String repoBaseUri, String groupId, String artifactId)
+  {
+    var uri = repoBaseUri + "/" + groupId.replace(".", "/") + "/" + artifactId + "/" + "maven-metadata.xml";
+    try
+    {
+      var client = HttpClient.newHttpClient();
+      var request = HttpRequest.newBuilder(URI.create(uri)).build();
+      var response = client.send(request, BodyHandlers.discarding());
+      assertThat(response.statusCode()).as("maven artifact seems to not exist " + uri).isEqualTo(200);
```

Comment:
cool @alexsuter, but shouldn't we validate the 'version' too?

Reply:
there is no version in meta.json
this will be calculated on the website at runtime. the version is backed by the maven artifactory itself. if we have problems there we need to fix this on the website.

Reply:
ahh true, thanks for the clarification